From: Mathias Kresin <dev@kresin.me>
Date: Sat, 1 Apr 2017 10:46:44 +0200
Subject: ramips: rename Digineo AC1200 Pro to ZBT-WG3526 32MB

The Digineo AC1200 Pro is the 32MB flash variant of the ZBT-WG3526 with
unpopulated/exposed sdhci slot. Rename to board to the OEM/ODM name and
add the sdhci kernel module to use it for multiple clones.

Signed-off-by: Mathias Kresin <dev@kresin.me>

Conflicts (Resolved):
	target/linux/ramips/base-files/etc/diag.sh

Signed-off-by: Vincent Wiemann <me@bibbl.com>

further adjustments for backport to lede-17.01 branch

Signed-off-by: Andreas Ziegler <github@andreas-ziegler.de>

diff --git a/target/linux/ramips/base-files/etc/board.d/02_network b/target/linux/ramips/base-files/etc/board.d/02_network
index 5a4042339c6761703578e014612062ac78babd9d..b9c8fe774fedab34267a8635033b9235d9d6c3aa 100755
--- a/target/linux/ramips/base-files/etc/board.d/02_network
+++ b/target/linux/ramips/base-files/etc/board.d/02_network
@@ -70,7 +70,6 @@ ramips_setup_interfaces()
 		ucidef_set_interface_lan "eth0.2"
 		;;
 	3g-6200n|\
-	ac1200pro|\
 	ai-br100|\
 	d240|\
 	db-wrt01|\
diff --git a/target/linux/ramips/base-files/etc/diag.sh b/target/linux/ramips/base-files/etc/diag.sh
index 9dbd6c4141d7facef2108cbb5cf3a7057522c88e..c4f521954c74635c61f42558f0d9b71f0f671c22 100644
--- a/target/linux/ramips/base-files/etc/diag.sh
+++ b/target/linux/ramips/base-files/etc/diag.sh
@@ -59,7 +59,10 @@ get_status_led() {
 	wn3000rpv3)
 		status_led="$board:red:power"
 		;;
-	ac1200pro|\
+	ai-br100|\
+	ht-tm02)
+		status_led="$board:blue:wlan"
+		;;
 	all0239-3g|\
 	dcs-930|\
 	dir-300-b1|\
@@ -85,10 +88,6 @@ get_status_led() {
 	zbt-wg3526)
 		status_led="$board:green:status"
 		;;
-	ai-br100|\
-	ht-tm02)
-		status_led="$board:blue:wlan"
-		;;
 	atp-52b|\
 	ip2202)
 		status_led="$board:green:run"
diff --git a/target/linux/ramips/base-files/lib/ramips.sh b/target/linux/ramips/base-files/lib/ramips.sh
index 8e27a79fc1744d6bed9db8d930207c418899cab1..b9279a5a1e9baf55ca7f59dbb22ac0b0b7b01db4 100755
--- a/target/linux/ramips/base-files/lib/ramips.sh
+++ b/target/linux/ramips/base-files/lib/ramips.sh
@@ -130,9 +130,6 @@ ramips_board_detect() {
 	*"DCS-930L B1")
 		name="dcs-930l-b1"
 		;;
-	*"Digineo AC1200 Pro")
-		name="ac1200pro"
-		;;
 	*"DIR-300 B1")
 		name="dir-300-b1"
 		;;
@@ -640,6 +637,9 @@ ramips_board_detect() {
 	*"ZBT-WG3526 (16M)")
 		name="zbt-wg3526-16M"
 		;;
+	*"ZBT-WG3526 (32M)")
+		name="zbt-wg3526-32M"
+		;;
 	*"ZBT-WR8305RT")
 		name="zbt-wr8305rt"
 		;;
diff --git a/target/linux/ramips/base-files/lib/upgrade/platform.sh b/target/linux/ramips/base-files/lib/upgrade/platform.sh
index d3efc2dd37ed855a1c0e966e448b67ae73324d5f..a45275077df8fa317c38adc8f0ab12fd4b0ecb83 100755
--- a/target/linux/ramips/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ramips/base-files/lib/upgrade/platform.sh
@@ -17,7 +17,6 @@ platform_check_image() {
 	3g150b|\
 	3g300m|\
 	a5-v11|\
-	ac1200pro|\
 	ai-br100|\
 	air3gii|\
 	all0239-3g|\
diff --git a/target/linux/ramips/dts/AC1200pro.dts b/target/linux/ramips/dts/AC1200pro.dts
deleted file mode 100644
index fbec81844025cf2a6544dec8dfeca02fce89fdc1..0000000000000000000000000000000000000000
--- a/target/linux/ramips/dts/AC1200pro.dts
+++ /dev/null
@@ -1,15 +0,0 @@
-/dts-v1/;
-
-#include "ZBT-WG3526.dtsi"
-
-/ {
-	model = "Digineo AC1200 Pro";
-};
-
-&firmware {
-	reg = <0x50000 0x1fb0000>;
-};
-
-&status_led {
-	label = "ac1200pro:green:status";
-};
diff --git a/target/linux/ramips/dts/ZBT-WG3526-16M.dts b/target/linux/ramips/dts/ZBT-WG3526-16M.dts
index fb644502071d6641194c66baf552035937f35669..1e2f1525028f69468257cd8af8d5bad0319bc551 100644
--- a/target/linux/ramips/dts/ZBT-WG3526-16M.dts
+++ b/target/linux/ramips/dts/ZBT-WG3526-16M.dts
@@ -9,7 +9,3 @@
 &firmware {
 	reg = <0x50000 0xfb0000>;
 };
-
-&status_led {
-	label = "zbt-wg3526:green:status";
-};
diff --git a/target/linux/ramips/dts/ZBT-WG3526-32M.dts b/target/linux/ramips/dts/ZBT-WG3526-32M.dts
new file mode 100644
index 0000000000000000000000000000000000000000..99a4b45824ba4ee6c1a1c81615291c8efde1a7ac
--- /dev/null
+++ b/target/linux/ramips/dts/ZBT-WG3526-32M.dts
@@ -0,0 +1,11 @@
+/dts-v1/;
+
+#include "ZBT-WG3526.dtsi"
+
+/ {
+	model = "ZBT-WG3526 (32M)";
+};
+
+&firmware {
+	reg = <0x50000 0x1fb0000>;
+};
diff --git a/target/linux/ramips/dts/ZBT-WG3526.dtsi b/target/linux/ramips/dts/ZBT-WG3526.dtsi
index 582e42083920c465588f14e59e10eafe735640fb..5e195fb8120d3a66f19acf83c9f76bb4b1ec3e7e 100644
--- a/target/linux/ramips/dts/ZBT-WG3526.dtsi
+++ b/target/linux/ramips/dts/ZBT-WG3526.dtsi
@@ -35,7 +35,8 @@
 	gpio-leds {
 		compatible = "gpio-leds";
 
-		status_led: status {
+		status {
+			label = "zbt-wg3526:green:status";
 			gpios = <&gpio0 24 1>;
 		};
 	};
diff --git a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
index cabc86b28eb040f0d1e58d917e6edc4a97e80db7..aa84d78b9e42259cbee7de6c052fa7ab500a040a 100644
--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -34,14 +34,6 @@ define Device/11acnas
 endef
 TARGET_DEVICES += 11acnas
 
-define Device/ac1200pro
-  DTS := AC1200pro
-  IMAGE_SIZE := $(ralink_default_fw_size_32M)
-  DEVICE_TITLE := Digineo AC1200 Pro
-  DEVICE_PACKAGES := kmod-usb3 kmod-ata-core kmod-ata-ahci
-endef
-TARGET_DEVICES += ac1200pro
-
 define Device/dir-860l-b1
   DTS := DIR-860L-B1
   BLOCKSIZE := 64k
@@ -235,6 +227,15 @@ define Device/zbt-wg3526-16M
 endef
 TARGET_DEVICES += zbt-wg3526-16M
 
+define Device/zbt-wg3526-32M
+  DTS := ZBT-WG3526-32M
+  IMAGE_SIZE := $(ralink_default_fw_size_32M)
+  SUPPORTED_DEVICES += ac1200pro
+  DEVICE_TITLE := ZBT WG3526 (32MB flash)
+  DEVICE_PACKAGES := kmod-usb3 kmod-usb-ledtrig-usbport kmod-ata-core kmod-ata-ahci kmod-sdhci-mt7620
+endef
+TARGET_DEVICES += zbt-wg3526-32M
+
 # FIXME: is this still needed?
 define Image/Prepare
 #define Build/Compile
